lib/stateTransition/AbstractStateTransition.js

```diff
@@ -49,6 +77,143 @@ class AbstractStateTransition {
   hash() {
     return hash(this.serialize()).toString('hex');
   }
+
+  /**
+   *  Returns signature
+   *
+   * @return {string|null}
+   */
+  getSignature() {
+    return this.signature;
+  }
+
+  /**
+   * Returns public key id
+   *
+   * @returns {number|null}
+   */
+  getSignaturePublicKeyId() {
+    return this.signaturePublicKeyId;
+  }
+
+  /**
+   * Sign data and check identityPublicKey
+   *
+   * @param {IdentityPublicKey} identityPublicKey
+   * @param {string|Buffer|Uint8Array|PrivateKey} privateKey string must be hex or base58
+   * @return {AbstractStateTransition}
+   */
+  sign(identityPublicKey, privateKey) {
+    const data = this.serialize({ skipSignature: true });
+    let privateKeyModel;
+    let pubKeyBase;
+
+    switch (identityPublicKey.getType()) {
+      case IdentityPublicKey.TYPES.ECDSA_SECP256K1:
+        privateKeyModel = new PrivateKey(privateKey);
+
```

Comment:
Please add comment that we are checking that public key is corresponding to the private.